src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -1292,23 +1292,23 @@ public static int ObjToHashAlgId(Object hashAlg)
                 if (hashAlg is SHA512)
                     return CapiHelper.CALG_SHA_512;
             }
-            else if (hashAlg is Type)
```

Comment:
True, and at the time I considered that since the only effect is to add a slight extra cost to a case that will throw anyway, it was of no odds, but I'll change to keep the flow under the principle of localising changes made.